### PR TITLE
Gemm Universal unit tests for MainloopIntelW8A8 API

### PR DESCRIPTION
**Pull request by rishi-yadav**
## Unit tests for MainloopIntelW8A8 FP8 GEMM operations on Intel Xe

Comprehensive test coverage for Intel GPU FP8 GEMM kernels including:
- **LLM workloads**: LLaMA2-7B, Mistral-7B configurations
- **Parallelization**: Tensor/model parallel scenarios  
- **Batch sizes**: Micro-batch (4x) to large batch (32x)
- **Matrix shapes**: Small (64²) to large (2048²), tall/wide matrices
- **Edge cases**: Large K/N dimensions

Tests validate `MainloopIntelW8A8` dispatch policy with FP8→FP32 precision on Intel Xe XMX architecture.

### CODE REVIEW

File: test/unit/gemm/device/gemm_testbed_3x.hpp [hunk 1]
```diff
@@ -4187,6 +4187,47 @@ bool TestXe(
...
+    passed = testbed.run(problem_size,
+                         cutlass::from_real<ElementScalar>(alpha),
+                         cutlass::from_real<ElementScalar>(beta));
+  }
+  catch (std::exception const& e) {
+    EXPECT_TRUE(false) << "TestXe: testbed.run threw an exception: " << e.what();
+    throw;
+  }
+  catch (...) {
+    EXPECT_TRUE(false) << "TestXe: testbed.run threw an unknown exception";
```
[the remark this answers is not in the document]
Comment (rolandschulz) [reply test/unit/gemm/device/gemm_testbed_3x.hpp]: copilot is wrong with `FAIL`. `ADD_FAILURE` might be slightly better but I'm OK either way.
